src/shared/seccomp-util.c

```diff
@@ -873,17 +873,42 @@ int seccomp_protect_sysctl(void) {
 }
 
 int seccomp_restrict_address_families(Set *address_families, bool whitelist) {
-
-#if !SECCOMP_RESTRICT_ADDRESS_FAMILIES_BROKEN
```

Comment:
this is actually on purpose!

The syscall ABI our own program uses is clear at compile time, and we can hence deduce easily with compile-time macros. However the available syscall ABIs in addition to the one we are using ourselves are something more complicated, is not as obvious, and we hence use the seccomp apis for that.

Hence: I conditioned this differently in the implementation and in the test, fully on purpose.